source/module_io/numerical_basis_jyjy.cpp

```diff
@@ -0,0 +1,96 @@
+#include "module_io/numerical_basis_jyjy.h"
+#include "module_basis/module_nao/two_center_integrator.h"
+
+namespace NumericalBasis
+{
+#ifdef __LCAO
+using index_t = std::tuple<int, int, int, int>;
+
+std::vector<index_t> indexgen(const std::vector<int>& natom,
+                              const std::vector<int>& lmax)
+{
+#ifdef __DEBUG
+    assert(natom.size() == lmax.size());
+#endif
+    std::vector<index_t> index;
+    for (size_t itype = 0; itype < natom.size(); ++itype)
+    {
+        for (int iatom = 0; iatom < natom[itype]; ++iatom)
+        {
+            for (int l = 0; l <= lmax[itype]; ++l)
+            {
+                for (int M = 0; M < 2*l+1; ++M)
+                {
+                    // convert the "abacus M" to the conventional m
+                    int m = (M % 2 == 0) ? -M/2 : (M+1)/2;
+                    index.emplace_back(itype, iatom, l, m);
+                }
+            }
+        }
+    }
+    return index;
+}
+
+
+ModuleBase::ComplexArray cal_overlap_Sq(
+    const char type,
+    const int lmax,
+    const int nbes,
+    const double rcut,
+    const std::vector<std::vector<ModuleBase::Vector3<double>>>& R,
+    const std::vector<index_t> mu_index
+)
+{
+    // allocate output array
+    int nlocal = mu_index.size();
+    ModuleBase::ComplexArray overlap_Sq(nlocal, nlocal, nbes, nbes);
+    overlap_Sq.zero_out();
+
+    // build a RadialCollection of spherical Bessel functions
+    double dr = 0.005; // grid spacing for SphbesRadials; smaller for higher precision
```

Comment:
const double dr ?

Reply:
A const qualifiers has been added in the latest commit.